iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/HelmInputFileContext.java

```diff
@@ -81,11 +82,18 @@ public String getSourceWithComments() {
   }
 
   public boolean isInChartRootDirectory() {
-    var rootChartDirectory = HelmFileSystem.retrieveHelmProjectFolder(Path.of(inputFile.uri()), sensorContext.fileSystem());
-    return inputFile.path().getParent() != null && inputFile.path().getParent().equals(rootChartDirectory);
+    return inputFile.path().getParent() != null && inputFile.path().getParent().equals(helmProjectDirectory);
   }
 
   public LinesShifting sourceMap() {
     return linesShifting;
   }
+
+  public Path getHelmProjectDirectory() {
+    return helmProjectDirectory;
+  }
+
+  public void setHelmProjectDirectory(Path helmProjectDirectory) {
+    this.helmProjectDirectory = helmProjectDirectory;
+  }
```

Comment:
I'm unsure if we should set this with a setter, or:

- as a constructor argument
- calculated in the constructor as we have all the information needed

I previously had a strong opinion to calculating it in the constructor. But seeing how other fields of this class are handled I'm unsure now.
WDYT?

Reply:
Yes, I agree that it makes sense to move this to the constructor. As for other fields, they are set closer to Helm evaluation. For additional files it can also makes sense to set them in the constructor, I'll create a ticket for that.